iree/turbine/kernel/wave/wave.py

```diff
@@ -231,6 +231,9 @@ def _trace_and_get_kernel_signature(
         # Set indices.
         set_node_indices(graph, self.constraints)
 
+        # Analyze Thread Shapes per Op.
+        determine_thread_shapes(graph)
```

Comment:
thanks! :)

Reply:
Actually thinking about this more, this could cause some issues with the symbol renaming I am doing. Just wondering does thread shape analysis really get easier by moving this here or can we keep it post expansion after I remove all the renaming?

Reply:
What I am thinking currently is rename, set index, expand, set post expand index, remove renames and then rest of pipeline.

Reply:
We can leave thread shape analysis here as well, but then it needs to handle renames and essentially remap variables based on the dicts in renames.

Reply:
yeah, I think this warrants a lower latency communication on VC haha